### PR TITLE
Azure.Core feature template

### PR DESCRIPTION
**Pull request by christothes**
Proposed Issue template for Azure.Core features.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Core Feature Request
```
Comment (jsquire) [inline .github/ISSUE_TEMPLATE/04_core_feature_request.yml:1]: If this is intended as something internal, maybe we should annotate that here?   Do we see a desire for customers to be making direct requests for Core features?

**Review by jsquire — APPROVED** [review]

**Review by annelo-msft — COMMENTED** [review]
Is there a way to visualize this so we can see what the UI will look like?

**Review by JoshLove-msft — APPROVED** [review]

**Comment by AlexGhiondea** [comment]
@annelo-msft I think you can switch the view in the review pane and see that rendered markdown for this PR.
image


I think it looks good!

**Review by AlexGhiondea — APPROVED** [review]

**Comment by annelo-msft** [comment]
Thank you @AlexGhiondea!!

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        - "Yes"
+        - "No"
+        - "Unsure"
+    validations:
+      required: true
+  - type: textarea
+    id: background
+    attributes:
+      label: Background and motivation
+      description: Please describe the purpose and value of the new feature and/or API here.
```
Comment (annelo-msft) [inline .github/ISSUE_TEMPLATE/04_core_feature_request.yml:23]: Could we make this more Azure SDK-specific by asking specifically for:
- What service libraries drove the requirement, and what the scenario is
- What service libraries will need to adopt the change

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      placeholder: Purpose
+    validations:
+      required: true
+  - type: textarea
+    id: api-proposal
+    attributes:
+      label: API / Feature Proposal
+      description: |
+        Please provide the specific public API signature diff that you are proposing and/or proposed feature designs
```
Comment (annelo-msft) [inline .github/ISSUE_TEMPLATE/04_core_feature_request.yml:32]: Could we suggest using APIView for this as an option?

**Review by annelo-msft — APPROVED** [review]
I love it, just a couple comment re: tweaks to descriptions.  Thanks for doing this, @christothes!

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml (hunk 3 above)
Comment (christothes) [reply .github/ISSUE_TEMPLATE/04_core_feature_request.yml:32]: The only downside to that is that it requires that the changes be stubbed out, and packaged in order to get them there. The changes may only be conceptual at this phase. But I think it is worth mentioning APIView.

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml (hunk 2 above)
Comment (christothes) [reply .github/ISSUE_TEMPLATE/04_core_feature_request.yml:23]: We could add it to this section https://github.com/Azure/azure-sdk-for-net/pull/27827/files#diff-4aecfafb7ad774892c793025c1a73e289e77e759c692d8477b2714f62b07fa1bR50

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml (hunk 3 above)
Comment (christothes) [inline .github/ISSUE_TEMPLATE/04_core_feature_request.yml:32]:
```suggestion
        Please provide the specific public API signature diff that you are proposing and/or proposed feature designs. Include a link to Api View if one exists. ```

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        }
+        ```
+    validations:
+      required: true
+  - type: textarea
+    id: constraints
+    attributes:
+      label: Time Constraints and Dependencies
+      description: Please outline any relevant dates related to this feature and reference any issues that are dependent on this feature
```
Comment (christothes) [inline .github/ISSUE_TEMPLATE/04_core_feature_request.yml:50]:
```suggestion
      description: Please outline any relevant deadline dates related to this feature. Please also list any client libraries that will consume these changes and describe the scenario this enables.```

File: .github/ISSUE_TEMPLATE/04_core_feature_request.yml (hunk 2 above)
Comment (christothes) [reply .github/ISSUE_TEMPLATE/04_core_feature_request.yml:23]: Added some more text based on your suggestions

**Pull request merged**